src/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore/DatabaseErrorPageMiddleware.cs

```diff
@@ -78,6 +78,8 @@ public void Hold(Exception exception, Type contextType)
             _options = options.Value;
             _logger = loggerFactory.CreateLogger<DatabaseErrorPageMiddleware>();
 
+            // Note: this currently leaks if the server hosting this middleware is disposed.
```

Comment:
Want to try the IMiddleware fix here?

Reply:
Depends what it is :trollface:

Reply:
Change the class to implement IMiddleware and IDisposable. The catch is that the class has to be registered as a service. Diagnostics does not currently register any services.

Reply:
What does registration of the middleware look like from UseDatabaseErrorPage?

Reply:
It's still `app.UseMiddleware<DatabaseErrorPageMiddleware>`, but passing in options directly might not work.